src/adapters/logx/execution/src/index.js

```diff
@@ -0,0 +1,55 @@
+const axios = require('axios');
+const ethers = require('ethers');
+
+async function getUserTVLData(blockNumber) {
+  const res = (await axios.get("https://backend-lp.logx.trade/zklink/tvl?blockNumber=" + blockNumber)).data;
+  const timestamp = await getTimestampAtBlock(blockNumber);
+  const tokenAddress = "0x46623d659cFDEf15cf5754C56bEc3F3f0B508fd8";
+  const poolAddress = "0x75940cDa18F14D1F97562fc2A6dBCe31CBe03870";
+  const symbol = 'LLP';
+
+  const returnArray = res.map((item) => {
+    return {
+      timestamp: timestamp,
+      userAddress: item.user,
+      tokenAddress: tokenAddress,
+      poolAddress: poolAddress,
+      balance: item.tvl,
+      symbol: symbol
+    }
+  })
+  return returnArray;
+};
+
+async function getUserTransactionData (startBlock, endBlock) {
+  const res = (await axios.get("https://backend-lp.logx.trade/zklink/volume?startBlock=" + startBlock + "&endBlock=" + endBlock)).data;
+  const contractAddress = "0x75940cDa18F14D1F97562fc2A6dBCe31CBe03870";
+  const tokenAddress = "0x2F8A25ac62179B31D62D7F80884AE57464699059";
+  const decimals = 6;
+  const price = 1;
+  const symbol = 'USDT';
+
+  const returnArray = res.map((item) => {
+    return {
+      timestamp: Math.floor(new Date(item.timestamp).getTime() / 1000),
+      userAddress: item.account,
+      contractAddress: contractAddress,
+      tokenAddress: tokenAddress,
+      decimals: decimals,
+      price: price,
+      quantity: Number(item.sizeDelta)/1e30,
+      txHash: item.txhash,
+      nonce: item.blockNumber,
```

Comment:
Please refer to the documentation. The nonce should be a unique value.

Reply:
No other issues 🎉

Reply:
> Please refer to the documentation. The nonce should be a unique value.

It should be a unique value for a transaction right? But in our architecture a single tx will never include multiple transfers, so I think having block number as nonce in this architecture should be fine as a single tx will never involve multiple trades, transfers or any of the operations related to TVL or Volume trading.

Reply:
I got two data entries with the same tx hash in CSV file
```
0xFfFb6B4E75c742E5DD61674a63b6D31AA3F826E6,0x75940cDa18F14D1F97562fc2A6dBCe31CBe03870,0x2F8A25ac62179B31D62D7F80884AE57464699059,6,1,260.01300000000003,0x28e3d2ca9dcbc095a5fc63d620543d3860cd37d040640395618cdf46bb2eced2,1772423,1716057866,

0x456Ff038e9BB0F8Cb457173d4c69a57ed9642906,0x75940cDa18F14D1F97562fc2A6dBCe31CBe03870,0x2F8A25ac62179B31D62D7F80884AE57464699059,6,1,260.01300000000003,0x28e3d2ca9dcbc095a5fc63d620543d3860cd37d040640395618cdf46bb2eced2,1772423,1716057866,
```